packages/cli/src/utils/utils.ts

```diff
@@ -113,27 +114,131 @@ export function findReplace(manifest: string, replacer: RegExp, value: string):
   return manifest.replace(replacer, value);
 }
 
+export function replaceArrayValueInTsManifest(manifest: string, key: string, newValue: string): string {
+  const start = manifest.indexOf(`${key}:`);
+  if (start === -1) return manifest; // If the value is not found, return the original manifest
+
+  let openBrackets = 0;
+  let startIndex, endIndex;
+
+  for (let i = start; i < manifest.length; i++) {
+    if (manifest[i] === '[') {
+      if (openBrackets === 0) startIndex = i;
+      openBrackets++;
+    } else if (manifest[i] === ']') {
+      openBrackets--;
+      if (openBrackets === 0) {
+        endIndex = i;
+        break;
+      }
+    }
+  }
+
+  if (openBrackets !== 0) return manifest; // Unbalanced brackets, return the original manifest
+
+  // Replace the old array with the new value
+  return manifest.slice(0, startIndex) + newValue + manifest.slice(endIndex + 1);
+}
+
+//  JavaScript's regex engine does not support recursive patterns like (?1).
+//  This regex would work in engines that support recursion, such as PCRE (Perl-Compatible Regular Expressions).
+export function extractArrayValueFromTsManifest(manifest: string, value: string): string | null {
+  const start = manifest.indexOf(`${value}:`);
+  if (start === -1) return null;
+
+  let openBrackets = 0;
+  let startIndex, endIndex;
+
+  for (let i = start; i < manifest.length; i++) {
+    if (manifest[i] === '[') {
+      if (openBrackets === 0) startIndex = i;
+      openBrackets++;
+    } else if (manifest[i] === ']') {
+      openBrackets--;
+      if (openBrackets === 0) {
+        endIndex = i;
+        break;
+      }
+    }
+  }
+
+  if (openBrackets !== 0) return null; // Unbalanced brackets
+
+  return manifest.slice(startIndex, endIndex + 1);
+}
```

Comment:
Pretty much a duplicate of the above function, see if you can generalise and improve code reuse